crates/biome_js_analyze/src/lint/nursery/no_vue_setup_props_reactivity_loss.rs

```diff
@@ -0,0 +1,775 @@
+                    return false;
+                };
+                string_literal.value_token().is_ok_and(|tok| {
+                    let text = tok.text_trimmed();
+                    if text.len() >= 2 {
+                        let inner = &text[1..text.len() - 1];
+                        inner == "setup"
+                    } else {
+                        false
+                    }
+                })
+            })
+        }
+        biome_js_syntax::AnyJsObjectMemberName::JsMetavariable(_) => false,
+    }
+}
+fn get_function_first_parameter(func: &SetupFunction) -> Option<AnyJsBindingPattern> {
+    match func {
+        SetupFunction::Function(any_func) => get_any_js_function_first_parameter(any_func),
+        SetupFunction::Method(method) => {
+            let params = method.parameters().ok()?;
+            let param = params.items().iter().next()?.ok()?;
+            let formal_param = param.as_any_js_formal_parameter()?;
+            formal_param.as_js_formal_parameter()?.binding().ok()
+        }
+    }
+}
+
+fn get_any_js_function_first_parameter(func: &AnyJsFunction) -> Option<AnyJsBindingPattern> {
+    match func {
+        AnyJsFunction::JsFunctionDeclaration(decl) => {
+            let params = decl.parameters().ok()?;
+            let param = params.items().iter().next()?.ok()?;
+            let formal_param = param.as_any_js_formal_parameter()?;
+            formal_param.as_js_formal_parameter()?.binding().ok()
+        }
+        AnyJsFunction::JsFunctionExpression(expr) => {
+            let params = expr.parameters().ok()?;
+            let param = params.items().iter().next()?.ok()?;
+            let formal_param = param.as_any_js_formal_parameter()?;
+            formal_param.as_js_formal_parameter()?.binding().ok()
+        }
+        AnyJsFunction::JsArrowFunctionExpression(arrow) => {
+            let params = arrow.parameters().ok()?;
+            match params {
+                biome_js_syntax::AnyJsArrowFunctionParameters::JsParameters(js_params) => {
+                    let param = js_params.items().iter().next()?.ok()?;
+                    let formal_param = param.as_any_js_formal_parameter()?;
+                    formal_param.as_js_formal_parameter()?.binding().ok()
+                }
+                biome_js_syntax::AnyJsArrowFunctionParameters::AnyJsBinding(binding) => {
+                    Some(AnyJsBindingPattern::AnyJsBinding(binding))
+                }
+            }
+        }
+        _ => None,
+    }
+}
+
+fn check_root_scope_destructuring(
+    setup_fn: &SetupFunction,
+    props_binding: &JsIdentifierBinding,
+    model: &SemanticModel,
+) -> Vec<Violation> {
+    let mut violations = Vec::new();
+
+    for reference in model.as_binding(props_binding).all_reads() {
+        if let Some(identifier) =
+            biome_js_syntax::JsReferenceIdentifier::cast(reference.syntax().clone())
```

Comment:
Remove the name of the crate, import the type at the beginning of the file.

Also, use `cast_ref` instead. Need to use `clone`